File: Lab9/javaScript/main.js
function main() {
    let arr = readLineOfWords();
    console.log("Слова з рядка:");
    arr.forEach(el => {
        console.log(el);
    });
    let resArr = deleteOddsAndFindMax(arr);
    console.log("Результат:");
    resArr.forEach(el => {
        console.log(el)
    })
}

function deleteOddsAndFindMax(arr) {
    let resArr = [];
    let max = 0;
    let maxIndex;
    for (let i = 0; i < arr.length; i += 2) {
        resArr[Math.trunc(i / 2)] = arr[i];
        if (arr[i].length > max) {
            max = arr[i].length;
            maxIndex = i / 2;
        }
    }
    console.log("Найбільший елемент:");
    console.log(resArr[maxIndex]);
    return resArr
}

function readLineOfWords() {
    let str = document.getElementById("inp1").value;
    return str.split(" ")
}

File: Lab9/golang/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	resSlice := readLineOfWords()
	fmt.Println("Список слів в рядку:")
	for _, el := range resSlice {
		fmt.Println(el)
	}
	resMap, maxIndex := deleteOddsAndFindMax(resSlice)
	fmt.Println("Результат:")
	for i := 0; i < len(resMap); i++ {
		fmt.Println(resMap[i])
	}
	fmt.Println("Найбільше слово:", resMap[maxIndex])
}
func deleteOddsAndFindMax(resSlice []string) (map[int]string, int) {
	resMap := make(map[int]string)
	var maxIndex = 0
	var max = 0
	for i := 0; i < len(resSlice); i += 2 {
		resMap[i/2] = resSlice[i]
		if len(resSlice[i]) > max {
			max = len(resSlice[i])
			maxIndex = i / 2
		}
	}
	return resMap, maxIndex
}
func readLineOfWords() []string {
	scanner := bufio.NewScanner(os.Stdin)
	for scanner.Scan() {
		s := scanner.Text()
		resSlice := strings.Split(s, " ")
		return resSlice
	}
	return nil
}
